tests/unit/-private/properties/focusable-test.js

```diff
@@ -0,0 +1,210 @@
+    assert.expect(1);
+
+    let page = create({
+      scope: '.scope',
+
+      foo: focusable('input')
+    });
+
+    this.adapter.createTemplate(this, page, '<div class="scope"><input /></div>');
+
+    this.adapter.$('.scope input').on('focus', () => assert.ok(1));
+
+    page.foo();
+
+    return this.adapter.wait();
+  });
+
+  test('resets scope', function(assert) {
+    assert.expect(1);
+
+    let page = create({
+      scope: '.scope',
+      foo: focusable('input', { resetScope: true })
+    });
+
+    this.adapter.createTemplate(this, page, '<input/>');
+
+    this.adapter.$('input').on('focus', () => assert.ok(1));
+
+    page.foo();
+
+    return this.adapter.wait();
+  });
+
+  test('returns target object', function(assert) {
+    assert.expect(1);
+
+    let page = create({
+      foo: focusable('input')
+    });
+
+    this.adapter.createTemplate(this, page, '<input/>');
+
+    this.adapter.$('input');
+
+    assert.equal(page.foo(), page);
+  });
+
+  test('finds element by index', function(assert) {
+    assert.expect(1);
+
+    let expectedSelector = 'input:eq(3)';
+    let page = create({
+      foo: focusable('input', { at: 3 })
+    });
+
+    this.adapter.createTemplate(this, page, '<input /><input /><input /><input />');
+
+    this.adapter.$(expectedSelector).on('focus', () => assert.ok(1));
+    page.foo();
+
+    return this.adapter.wait();
+  });
+
+  test('looks for elements outside the testing container', function(assert) {
+    assert.expect(1);
+
+    let expectedContext = '#alternate-ember-testing';
+    let page = create({
+      foo: focusable('input', { testContainer: expectedContext })
+    });
+
+    this.adapter.createTemplate(this, page, '<input />', { useAlternateContainer: true });
+
+    this.adapter.$('input', expectedContext).on('focus', () => assert.ok(1));
+
+    page.foo();
+
+    return this.adapter.wait();
+  });
+
+  test('looks for elements within test container specified at node level', function(assert) {
+    assert.expect(1);
+
+    let expectedContext = '#alternate-ember-testing';
+    let page = create({
+      testContainer: expectedContext,
+      foo: focusable('input')
+    });
+
+    this.adapter.createTemplate(this, page, '<input />', { useAlternateContainer: true });
+
+    this.adapter.$('input', expectedContext).on('focus', () => assert.ok(1));
+
+    page.foo();
+
+    return this.adapter.wait();
+  });
+
+  test("raises an error when the element doesn't exist", function(assert) {
+    assert.expect(1);
+
+    let page = create({
+      foo: {
+        bar: {
+          baz: {
+            qux: focusable('button')
+          }
+        }
+      }
+    });
+
+    this.adapter.createTemplate(this, page);
+
+    this.adapter.throws(assert, function() {
+      return page.foo.bar.baz.qux();
+    }, /page\.foo\.bar\.baz\.qux/, 'Element not found');
+  });
+
+  test('raises an error when the element is not focusable', function(assert) {
```

Comment:
I suppose to be really thorough, we would need another block like this with assertions confirming that elements that should be focusable don't throw an error.